src/junox/chorus.js

```diff
@@ -1,22 +1,31 @@
 import LFO from './lfo'
 
-const MAXBUFFERSIZE = 15000
+const MAXBUFFERSIZE = 256
 
 class RingBuffer {
   constructor() {
     this.buffer = new Float32Array(MAXBUFFERSIZE).fill(0)
     this.index = 0
   }
 
-  getSample(sampleIndex) {
-    let localIndex = this.index - sampleIndex
-    if (localIndex < 0) {
-      localIndex += MAXBUFFERSIZE
+  ringBufferIndex(index) {
+    if (index < 0) {
+      return index + MAXBUFFERSIZE
     }
-    if (localIndex >= MAXBUFFERSIZE) {
-      localIndex -= MAXBUFFERSIZE
+    if (index >= MAXBUFFERSIZE) {
+      return index - MAXBUFFERSIZE
     }
-    return this.buffer[localIndex]
+    return index
+  }
+
+  getSample(sampleIndex) {
+    let localIndex = this.ringBufferIndex(this.index - sampleIndex)
+    const fractional = localIndex - Math.abs(localIndex)
+    const indexA = Math.floor(localIndex)
+    const indexB = this.ringBufferIndex(Math.floor(localIndex + 1))
```

Comment:
const indexB = this.ringBufferIndex(indexA)

Reply:
good optimization idea!